deploifai/api/__init__.py

```diff
@@ -410,3 +409,213 @@ def create_project(self, project_name: str, cloud_profile: CloudProfile, fragmen
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                        "falconMLConfigId": falcon_id,
+                        "dataStorageIds": data_storage_id,
+                    },
+                }
+            else:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                        "falconMLConfigId": falcon_id,
+                    },
+                }
+        else:
+            if data_storage_id:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                        "dataStorageIds": data_storage_id,
+                    },
+                }
+            else:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                    },
+                }
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": mutation, "variables": variables},
+                headers=self.headers,
+            )
+
+            create_mutation_data = r.json()
+
+            return create_mutation_data["data"]["createTraining"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not create Training Server. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not create Training Server. Please try again.")
+
+    def get_server(self, workspace: str):
+        query = """
+        query ($whereAccount: AccountWhereUniqueInput!){
+          trainings(whereAccount: $whereAccount){
+            id
+            name
+            status
+          }
+        }
+        """
+
+        variables = {"whereAccount": {"username": workspace}}
+
+        r = requests.post(
+            self.uri,
+            json={"query": query, "variables": variables},
+            headers=self.headers,
+        )
+        server_details = r.json()["data"]["trainings"]
+        return server_details
```

Comment:
Should probably do some `try-except` here as well.

Reply:
Yes
Refactored